partcad/src/partcad/test/test.py

```
@@ -45,12 +49,14 @@
         message = self._log_message_prepare(*args)
         pc_logging.debug(f"Test: {shape.project_name}:{shape.name}: {self.name}{message}")
 
-    def failed(self, shape, *args) -> None:
+    def failed(self, shape, *args) -> bool:
```

Ruff (0.8.2)

Missing type annotation for function argument `shape`

(ANN001)